Content.Server/NPC/Systems/NPCImprintingOnSpawnBehaviourSystem.cs

```diff
@@ -0,0 +1,52 @@
+using System.Numerics;
+using Content.Shared.NPC.Components;
+using Content.Shared.NPC.Systems;
+using Robust.Shared.Map;
+using Robust.Shared.Random;
+
+namespace Content.Server.NPC.Systems;
+
+public sealed partial class NPCImprintingOnSpawnBehaviourSystem : SharedNPCImprintingOnSpawnBehaviourSystem
+{
+    [Dependency] private readonly EntityLookupSystem _lookup = default!;
+    [Dependency] private readonly NPCSystem _npc = default!;
+    [Dependency] private readonly IRobustRandom _random = default!;
+
+    public override void Initialize()
+    {
+        base.Initialize();
+        SubscribeLocalEvent<NPCImprintingOnSpawnBehaviourComponent, MapInitEvent>(OnMapInit);
+    }
+
+    private void OnMapInit(Entity<NPCImprintingOnSpawnBehaviourComponent> imprinting, ref MapInitEvent args)
+    {
+
+        var entities = _lookup.GetEntitiesInRange(imprinting, imprinting.Comp.SearchRadius);
+        var imprintingTargets = new List<EntityUid>();
+
+        if (entities == null) return;
+        if (entities.Count == 0) return;
```

Comment:
1. Entities can never be returned as null.
2. Use the overload that lets you pass in a set.

Reply:
idk what is Overload

Reply:
An overload is a method (function) that has the same name as another method, but takes different parameters. Like `Foo(string bar)` could have overloads like `Foo(int baz)` or `Foo(string bar, bool baz)` etc.

There's an overload for GetEntitiesInRange that takes a hashset as a parameter.